CMakeLists.txt

```diff
@@ -45,96 +76,75 @@ set(COMMON_WARNINGS
         -Wsign-compare
         -Wunused-parameter
         -Wpedantic
-        #-pedantic-errors
         -Wconversion
         -Wsign-conversion
         -Wreturn-type
         )
 # Clang-specific warnings compiler flags.
-set(CLANG_WARNINGS
+set(MATA_CLANG_WARNINGS
         -Winconsistent-missing-override
         -Wgnu-anonymous-struct
         -Wnested-anon-types
         -Wno-system-headers
         )
 
 # Set whether to treat warnings as errors.
-option(WERROR "WERROR" OFF)
-message("-- Treating warnings as errors: ${WERROR}")
-if (WERROR)
-    set(COMMON_WARNINGS "${COMMON_WARNINGS}" -Werror)
+if (MATA_WERROR)
+    message("-- Treating warnings in mata as errors")
+    set(MATA_COMMON_WARNINGS "${MATA_COMMON_WARNINGS}" -Werror)
 endif ()
 
-##############################################################################
-#                                COVERAGE
-##############################################################################
-
 # if DEBUG, also test coverage
-if(CMAKE_BUILD_TYPE MATCHES "Coverage")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -O0 -fprofile-arcs -ftest-coverage")
-    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -lgcov")
```

Comment:
Are you sure it is safe to remove the `-lgcov` flag for linker?

Reply:
Both local and GA coverage seems to work, so... It should be fine, it seems.

Reply:
From what I read `-lgcov` is needed only for `C` and not for `C++`. I have no idea why, but it seems to be working.

Reply:
Ook, if codecov works, I'm fine with that.